packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1107,11 +1107,15 @@ export class TransactionController extends BaseController<
         },
       );
 
+      // Ensure transaction parameters are still valid before adding metadata to state.
+      validateTxParams(addedTransactionMeta.txParams, isEIP1559Compatible);
```

Comment:
Since we already have a `validateTxParams` inside `updateTransactionInternal`, would it be even more effective to put this inside `addMetadata`?